RFC-0009-jdbc-join-push-down.md

```diff
@@ -0,0 +1,570 @@
+
+## JdbcJoinPushdown optimizer
+
+JoinPushdown Optimizer is implemented inside the presto-base-jdbc module. This optimizer is called after GroupInnerJoinsByConnector. It is used to convert JoinTableSet to List of ConnectorTableHandles which is able to be understood by JdbcTableHandle.
+
+- JdbcJoinPushdown Optimizer is added as Logical Plan Optimizer in JdbcPlanOptimizerProvider.
+```
+@Override
+public Set<ConnectorPlanOptimizer> getLogicalPlanOptimizers()
+{
+ return ImmutableSet.of(new JdbcJoinPushdown());
+}
+```
+- When the logical optimization of Jdbc connector happens, it invokes JdbcJoinPushdown optimizer visitTableScan() to rewrite JoinTableSet to List of ConnectorTableHandles
+
+Inside the visitTableScan() :
+- We check if connectorHandle of tableHandle is an instance of JoinTableSet
+- If that is the case, make a new JdbcTableHandle with joinTables as the tableHandles.getConnectorTableHandles()
+- If not, return the node.
+
+![JdbcJoinPushdown optimizer](RFC-0009-jdbc-join-push-down/After_JdbcJoinPushdown.png)
+
+JdbcJoinPushdown optimizer will create a TableScanNode structure which is able to hold all the jdbc tables which are grouped as part of above implementation. Below is the proposed structure for the new TableScanNode
+
+![Image 2](RFC-0009-jdbc-join-push-down/in-depth-design-image-2.png)
+
+## PredicatePushDown Optimizer
+
+#### Pushdown the overall filter to the newly created TableScanNode.
+After creating Single TableScanNode for grouped tables (refer point 7) we need to pushdown the FilterNode (join criteria specific to the grouped tables of new tableScanNode and all filters specific to the group tables) on connector level for the applicable filter and maintain the FilterNode for presto if it is not able to pushdown. For this we are just invoking predicate pushdown after jdbc join pushdown optimizer and there is no code change.
+
+## JdbcComputePushdown Optimizer
+Using JdbcComputePushdown optimizer, we are pushing down the join criteria as additional predicate. For doing this, we enhanced JdbcComputePushdown optimizer by adding a join predicate to sql translator. 
+```
+private final JdbcJoinPredicateToSqlTranslator jdbcJoinPredicateToSqlTranslator;
+```
+
+```
+this.jdbcJoinPredicateToSqlTranslator = new JdbcJoinPredicateToSqlTranslator(
+                functionMetadataManager,
+                buildFunctionTranslator(ImmutableSet.of(JoinOperatorTranslators.class)),
+                identifierQuote);
+```
+We have also added visitPlan() method and enhanced the visitFilter() method.
+
+Enhancements done to visitFilter() are from this PR : https://github.com/prestodb/presto/pull/16412/
+Before, if all the filter could not be pushed down, nothing would be pushed down.
+The changes help in finding what all can be pushed down and able to be translated. The filters that can't be pushed down are kept in a new FilterNode.
+
+## Changes in SPI module 
+
+- Added 2 new classes : JoinTableInfo, JoinTableSet
```

Comment:
Can you add a basic class outline for `JoinTableInfo` and `JoinTableSet` ? Just the members and any inheritance relationships would suffice